lib/index.js

```diff
@@ -60,10 +62,19 @@ function gitUrlParse(url) {
             break;
         default:
             splits = urlInfo.name.split("/");
-            if (splits.length === 2) {
+            if (splits.length > 2) {
                 urlInfo.owner = splits[0];
                 urlInfo.name = splits[1];
             }
+
+            urlInfo.ref = "";
+            urlInfo.filepath = "";
+            if ((splits.length > 3) & (["blob", "tree"].indexOf(splits[2]) != -1)) {
```

Comment:
Instead of `indexOf(...) != -1` you can do: `["blob", "tree"].includes(splits[2])`.

Also, I think `&` is a typo (unless you really meant it), I assume it should be `&&`.

Maybe it would be helpful to add the type to the urlInfo (`tree` or `blob`).

 :grin:

```js
// Find a better property name maybe
urlInfo.type = splits[2]
```

Nice work overall! :+1:

Reply:
👍 to comments, I think they're all addressed in latest push (+ a test).

LMKWYT!